internal/params/json.go

```diff
@@ -148,7 +151,22 @@ func makeSettingsFromObject(common commonSettings, rawSettings json.RawMessage,
 	if common.Provider == "google" {
 		return nil, nil, fmt.Errorf("%w: %s", ErrProviderNoLongerSupported, common.Provider)
 	}
-
+  
+  FQDN := ""
+  switch {
+    case (common.Host != "" && common.Host != "@") && common.Domain != "":
+      FQDN = common.Host + "." + common.Domain
+    case (common.Host == "@" || common.Host == "") && common.Domain != "":
+      FQDN = common.Domain
+    case common.Host != "" && common.Host != "@" && common.Domain == "":
+      FQDN = common.Host + ".fakedomain.com"
+  }
```

Comment:
Let's focus on validating the domain field for now only (what the PR and issue are for).
The host can be `@`, `*` or a valid subdomain (I'll hunt for an RFC precising what's the allowed characters etc.), but let's do this in a subsequent PR.

Reply:
The now included module does this well now that I've poked at it, just need to account for these special cases where the host won't RFC compliant.